CONTRIBUTING.md

```diff
@@ -103,3 +97,156 @@ The content of the commit message body should contain:
 
 - information about the SHA of the commit being reverted in the following format: `This reverts commit <SHA>`,
 - a clear description of the reason for reverting the commit message.
+
+# Instructions for Building and Running Locally
+
+This is a starting point for building Grafana Data Source Backend Plugins
+
+## What is Grafana Data Source Backend Plugin?
+
+Grafana supports a wide range of data sources, including Prometheus, MySQL, and even Datadog. There’s a good chance you can already visualize metrics from the systems you have set up. In some cases, though, you already have an in-house metrics solution that you’d like to add to your Grafana dashboards. Grafana Data Source Plugins enables integrating such solutions with Grafana.
+
+For more information about backend plugins, refer to the documentation on [Backend plugins](https://grafana.com/docs/grafana/latest/developers/plugins/backend/).
+
+## Getting started
+
+A data source backend plugin consists of both frontend and backend components.
+
+### Setup Plugin Directory
+
+1. Create directory:
+
+   ```bash
+   mkdir -p path/to/your/clone
+   ```
+
+2. Clone this repo into the new directory and update to `develop` branch:
+
+   ```bash
+   cd path/to/your/clone
+   git clone git@github.com:golioth/grafana-websocket-plugin.git .
+   git checkout develop
+   ```
+
+### Frontend
+
+1. Install dependencies
+
+   ```bash
+   yarn install
+   ```
+
+2. Build plugin in development mode or run in watch mode
+
+   ```bash
+   yarn dev
+   ```
+
+   or
+
+   ```bash
+   yarn watch
+   ```
+
+3. Build plugin in production mode
+
+   ```bash
+   yarn build
+   ```
+
+   - **Yarn Error**: If yarn throws `opensslErrorStack`, export this value in the terminal and build yarn again:
+
+     ```bash
+     export NODE_OPTIONS=--openssl-legacy-provider
+     ```
+
+### Backend
+
+1. Update [Grafana plugin SDK for Go](https://grafana.com/docs/grafana/latest/developers/plugins/backend/grafana-plugin-sdk-for-go/) dependency to the latest minor version:
+
+   ```bash
+   go get -u github.com/grafana/grafana-plugin-sdk-go
+   go mod tidy
+   ```
+
+2. Build backend plugin binaries for Linux, Windows and Darwin:
+
+   ```bash
+   mage -v
+   ```
+
+3. List all available Mage targets for additional commands:
+
+   ```bash
+   mage -l
+   ```
+
+### Run the Grafana
+
+1. Start the docker container
+
+   ```bash
+    docker run \
+      --network="host" -e "GF_LOG_MODE=console file" \
+      -e "GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS=golioth-websocket-datasource" \
+      -p 3000:3000 \
+      -v ~/Golioth/Codes/grafana-plugins:/var/lib/grafana/plugins \
```

Comment:
this probably need to be changed to `path/to/your/clone`

Reply:
Now it's ok.